safety_shield/include/safety_shield/robot_reach.h

```diff
@@ -260,6 +260,13 @@ class RobotReach {
     return cross;
   }
 
+  /**
+   * @brief returns robot capsules for velocity
+   * @return robot capsules for velocity
+   */
+  inline std::vector<occupancy_containers::capsule::Capsule>& getRobotCapsulesForVelocity() {
```

Comment:
I would not return a reference. Either return a pointer to the vector or the vector itself.
So 
`inline std::vector<occupancy_containers::capsule::Capsule> getRobotCapsulesForVelocity() {`
or 
 `inline std::vector<occupancy_containers::capsule::Capsule>* getRobotCapsulesForVelocity() {`